lib/dtutils/file.lua

```diff
@@ -0,0 +1,523 @@
+--[[
+
+    dtutils/file.lua - common darktable lua file functions
+
+    Copyright (C) 2016 Bill Ferguson <wpferguson@gmail.com>.
+    Copyright (C) 2016 Tobias Jakobs
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+]]
+
+local dtutils_file = {}
+dtutils_file.libdoc = {
+  Sections = {"Name", "Synopsis", "Description", "License"},
+  Name = [[dtutils.file - common darktable lua file functions]],
+  Synopsis = [[local df = require "lib/dtutils.file"]],
+  Description = [[dtutils.file provides common file manipulation functions used in
+  constructing Darktable lua scripts]],
+  License = [[This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 3 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.]],
+  functions = {}
+}
+
+local dt = require "darktable"
+
+local log = require "lib/libLog"
+
+local gettext = dt.gettext
+
+dt.configuration.check_version(...,{3,0,0})
+
+-- Tell gettext where to find the .mo file translating messages for a particular domain
+gettext.bindtextdomain("dtutils.file",dt.configuration.config_dir.."/lua/locale/")
```

Comment:
again, I don't really like that way of doing things... that place won't be exactly where the lua-scripts was cloned, so chances are high that the user has to move files out of the clone to the proper location, and do it every time he updates the scripts

use the "dump" functionnality of dt.debug to see exactly what's in the ... argument provided to your script and find the proper directory that way... that will be way more future-proof and also a great example for futur users of this library